lib/ecto/schema.ex

```diff
@@ -754,6 +780,12 @@ defmodule Ecto.Schema do
     Ecto.Model.Callbacks.__apply__(struct.__struct__, :after_load, loaded)
   end
 
+  def __load__(struct, source, fields, map, id_types) do
```

Comment:
This will also help with mongo driver, where we already have a map. Is there some more efficient way to handle this?